code/modules/buildmode/submodes/proccall.dm

```diff
@@ -0,0 +1,47 @@
+/datum/buildmode_mode/proccall
+	key = "proccall"
+	var/proc_name = null
+	var/list/proc_args = null
+
+/datum/buildmode_mode/proccall/show_help(client/target_client)
+	to_chat(target_client, span_notice("***********************************************************\n\
+		Right Mouse Button on buildmode button = Choose procedure and arguments\n\
+		Left Mouse Button on machinery = Apply procedure on object.\n\
+		***********************************************************"))
+
+/datum/buildmode_mode/proccall/change_settings(client/target_client)
+	if(!check_rights_for(target_client, R_DEBUG))
+		return
+
+	proc_name = input("Proc name, eg: fake_blood","Proc:", null) as text|null
+	if(!proc_name)
+		return
+
+	proc_args = target_client.get_callproc_args()
+	if(!proc_args)
+		return
+
+/datum/buildmode_mode/proccall/handle_click(client/target_client, params, datum/object as null|area|mob|obj|turf)
+	if(!proc_name || !proc_args)
+		tgui_alert(target_client, "Undefined ProcCall or arguments.")
+		return
+	
+	if(!hascall(object, proc_name))
+		to_chat(target_client, "<font color='red'>Error: callproc_datum(): type [object.type] has no proc named [proc_name].</font>", confidential = TRUE)
```

Comment:
```suggestion
		to_chat(target_client, "span_warning(Error: callproc_datum(): type [object.type] has no proc named [proc_name].)", confidential = TRUE)
```

No need to manually set the color when we have spans that do the heavy lifting for us.